### ISSUE TITLE
Make Istio conformance tests non-disruptive

### ISSUE THREAD

**Issue by johscheuer** [istio/istio#16615]
**Bug description**

Currently we are evaluate if the istio [conformance tests](https://github.com/istio/istio/tree/master/tests/integration/conformance) can be used to check our installation. What we discovered (it took me a way longer than I would like to confess) is that the conformance tests are disruptive and can stop the whole cluster from working. Currently we have globally activated [mTLS](https://istio.io/docs/tasks/security/authn-policy/#globally-enabling-istio-mutual-tls) and the conformance test will delete the default config in a test case (I didn't saw which one) and never recreates it. Currently there are two main issues here:

- At least the default configuration should be restored after the tests (yeah here is a todo: https://github.com/istio/istio/blob/master/tests/integration/conformance/conformance_test.go#L102)
- Anyway deleting/modifying the default `MeshPolicy` will probably interference with the rest of the Istio setup which means the conformance tests can never be run in a production cluster (actually this is something we want to do). A good example are the kubernetes conformance tests. So from my pov the conformance test should only use `Policies` and not the `MeshPolicy` if required.

If this is also a ~~problem~~ challenge from your pov for the Istio conformance tests I'm open to help where I can.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[X] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

I expected that I can run the conformance tests without breaking my cluster.

**Steps to reproduce the bug**

Create a default `MeshPolicy` to activate [globally mTLS](https://istio.io/docs/tasks/security/authn-policy/#globally-enabling-istio-mutual-tls) and run the istio conformance test.

After this the default  `MeshPolicy` is missing and all mTLS communication is broken (since the services will communicate over bare http).

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
1.2.4

**How was Istio installed?**
helm (and some custom patches for PSP)

**Environment where bug was observed (cloud vendor, OS, etc)**
bare-metal / Container Linux

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by johscheuer** [comment]
TLDR at least there should be a big fat warning that the conformance test can break your istio setup.

**Comment by ericvn** [comment]
Note that the integration test readme does state: **Be aware that any existing content will be altered and/or removed from the cluster.**

**Comment by johscheuer** [comment]
I didn't see this note but you are right the docs actually state this already. I would prefer to move this "warning" more at the top of the docs to be more prominent.

The main from my point of view is that conformance tests shouldn't be disruptive (like the Kubernetes ones) otherwise you will never be able to run a conformance test on a running cluster which was our idea.
